Pcap++/src/PcapDevice.cpp

```diff
@@ -20,7 +20,7 @@ bool IPcapDevice::setFilter(std::string filterAsString)
 		return false;
 	}
 
-	struct bpf_program prog;
+	struct bpf_program prog = {};

```

Comment:
This type of initialization is a C++11 feature. While I plan to move PcapPlusPlus to C++11 soon, as of now I wouldn't want to add C++11 specific code.

Reply:
OK, fair enough.  What do you suggest as an alternative, I'm not well-versed in C++?  Maybe explicitly setting the prog.bf_len = 0 and prog.bf_insns = NULL?

Reply:
I'm not sure this struct needs to be initialized. I think libpcap initializes it internally: https://www.tcpdump.org/pcap.html

Reply:
It needs to be initialized because if the compilation of the filter fails, your code attempts to free the bpf_program structure, but if you review the code for [pcap_compile](https://github.com/the-tcpdump-group/libpcap/blob/bb02779a37fdf815da58e8828eb4ce99a4f86580/gencode.c#L724), it initializes the bpf_program structure only at the conclusion of the function (when the filter compilation succeeds).  In a filter-compilation-failure case, it is not initialized, resulting in a wild pointer that pcap_freecode attempts to free.  A simple test that can demonstrates this is to set the prog.bf_len to an arbitrary value and then check its value after pcap_compile fails (by specifying an invalid filter), you'll note that the value is unchanged.

Reply:
So maybe we should just remove this call to `pcap_freecode`?

Reply:
lol, fair enough.

Reply:
@nickamon would you like to make this change so we can merge the PR?

Reply:
Hmm, kinda of messed up my branch, let me know if you want me to close this pull request and reopen a new one w/ just the fix.